mis/solver/greedymapping.py

```diff
@@ -0,0 +1,261 @@
+from __future__ import annotations
+
+import math
+import random
+from statistics import mean
+from typing import Dict, List, Set, Tuple
+
+import networkx as nx
+
+from mis.shared.types import MISInstance
+from mis.solver.lattice import Lattice
+
+class GreedyMapping:
+    """
+    Performs a greedy mapping of a MISInstance's graph onto a physical lattice layout.
+    """
+
+    def __init__(
+        self,
+        instance: MISInstance,
+        lattice: Lattice,
+        previously_generated_subgraphs: List[Dict[int, int]],
+        seed: int = 0,
+    ) -> None:
+        """
+        Initializes the GreedyMapping algorithm for mapping a graph onto a lattice.
+
+        Args:
+            instance: The MIS problem instance containing the logical graph.
+            lattice: The lattice structure defining the physical layout.
+            previous_mappings: List of previous mappings (for scoring reuse).
```

Comment:
Nit: `previous_mappings` or `previously_generated_subgraphs`? Also, reading this documentation, I don't understand what it is.